test/NetTopologySuite.Samples.Console/Technique/SplitPolygonAtDateline.cs

```diff
@@ -0,0 +1,802 @@
+﻿using NetTopologySuite.Densify;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.Geometries.Utilities;
+using NetTopologySuite.IO;
+using NetTopologySuite.Operation.Polygonize;
+using NetTopologySuite.Operation.Valid;
+using NetTopologySuite.Precision;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace NetTopologySuite.Samples.Technique
+{
+    /// <summary>
+    /// Presents a technique for converting coordinates to a polygon. With the coordinates all in one hemisphere it is a fairly simple process.
+    /// However, when the geometry crosses the dateline it becomes a more challenging problem. This routine attempts to do both. That is to say,
+    /// it converts coordinates to a poly geometry regardless if the geometry traverses the dateline or not. If the geometry does cross the dateline
+    /// then it is split up into multiple parts. This benefits the user in two ways: first, any GIS will be able to display the geometry in a proper
+    /// way regardless of map projection and two, the GIS will be able to perform geoprocessing functions on the geometry. Note that standard feature
+    /// classes do not support individual features with multiple parts that share the same edge. Thus, it is left up to the user on how to commit
+    /// the geometries to a feature class. Suggestions are to save the multiple geometries as individual features or to slightly alter the edges of
+    /// each geometry so as to save all parts as a single multi-part feature. For the latter option, the algorithm provides an input to trim edges
+    /// of parts so that they can be used in a single feature.
+    /// </summary>
+    public class SplitPolygonAtDateline
+    {
+        [STAThread]
+        public static void main()
+        {
+            // ========================================================================================
+            // inputs
+            // ========================================================================================
+            string wktPrj = "GEOGCS['WGS 84',DATUM['WGS_1984',SPHEROID['WGS 84',6378137,298.257223563,AUTHORITY['EPSG','7030']],AUTHORITY['EPSG','6326']]," +
```

Comment:
From what I understand, the projection string is used just ot check if the data is in a GEOGCS format. To me this check is useless, because we are not checking the geometry but a separate parameters. Maybe a check of the geometry values (inside the valid range of values) can be slower but useful?

Reply:
I suppose I can do that but say if someone has coordinates in UTM that go from 50 to 170 meters. That will pass the geographic check as we are only checking values and not units. My thought was that I didn't want users to just hand over their coordinates hoping it would work with any projection. The easiest way, for me at least, was to force them to provide a projection string.

Reply:
I see your point, but maybe it's something related to the documentation of the method/api. my2cents

Reply:
Absolutely. I strived to add quite a bit of a description buttttttt there will be many that don't read it and just dive in. :)